Sources/RInAppMessaging/Models/Responses/Campaign.swift

```diff
@@ -52,10 +52,10 @@ internal struct Campaign: Codable, Hashable {
         try container.encode(isOptedOut, forKey: .isOptedOut)
     }
 
-    init(data: CampaignData) {
+    init(data: CampaignData, asTooltip isTooltip: Bool = true) {
```

Comment:
I don't like this new parameter. It's just aborting tooltip data parsing making the logic unnecessarily complicated just for unit tests. Please remove that flag and use either invalid tooltip data payload or just a normal campaign to make `tooltipData` nil

Reply:
I agree with you, but now the only way to make Campaign instance is by passing **CampaignData to the constructor.**
Invalid tooltip data payload cannot be achieved because TooltipData inside the CampaignData is required and never be nil.

Reply:
@imaginaris 
I remove the additional params in the constructor on this commit [1ca5512](https://github.com/rakutentech/ios-inappmessaging/pull/275/commits/1ca55121edaadfeb99f5463f0caf2a453f7aba1c).

Since the `MessagePayload` is required and will never be nil. I am thinking why the TooltipData is `optional?` because it will always have data even if the MessagePayload is invalid.

The solution I am thinking of is like the below approach:

1. Remove the optional in this code.
```swift
// Campaign.swift | L.12
let tooltipData: TooltipData?
```

2. Add a new constructor for tooltip data.
```swift
    init(data: CampaignData) {
        self.data = data
        impressionsLeft = data.maxImpressions
        tooltipData = nil
    }

    init(data: CampaignData, tooltipData: MessagePayload) {
        self.data = data
        impressionsLeft = data.maxImpressions
        self.tooltipData = Campaign.parseTooltipData(messagePayload: tooltipData)
    }
```

Reply:
Hmm how can you remove optional from here `let tooltipData: TooltipData?` and then have this line in the constructor: `tooltipData = nil`?
`Campaign.parseTooltipData(messagePayload: tooltipData)` can return nil and it will when message body doesn't contain a valid json file that can be mapped to `TooltipData` model.
The only way to distinguish tooltip payload from normal campaign payload is by checking the content of `messageBody`. Anyway, the main point is that it's a temporary measure. The tooltip feature is in an alpha phase. The data model parsing will change when IAM Dashboard start supporting tooltip campaign type.

Reply:
> Invalid tooltip data payload cannot be achieved because TooltipData inside the CampaignData is required and never be nil.
What do you mean by 'TooltipData inside the CampaignData'?
You can have invalid tooltip data payload by putting an empty string in `messageBody` or a json with a missing key the is required.
(ref: https://github.com/rakutentech/ios-inappmessaging/blob/master/Tests/Tests/SerializationSpec.swift#L126)

Reply:
Yes, you are correct and thank you for sending the ref. I finally can achieve the invalid tooltipData with `""` messageBody. I update the code here [1277742](https://github.com/rakutentech/ios-inappmessaging/pull/275/commits/127774286b2c453a2a3c279fc8a1d61dcb388b3c).